src/ert/gui/tools/plot/plot_case_selection_widget.py

```diff
@@ -1,109 +1,81 @@
-        self.__add_case_button.setIcon(QIcon("img:add_circle_outlined.svg"))
-        self.__add_case_button.setEnabled(len(self._cases) > 0)
-        self.__add_case_button.clicked.connect(self.addCaseSelector)
-
-        add_button_layout.addStretch()
-        add_button_layout.addWidget(self.__add_case_button)
-        add_button_layout.addStretch()
-
-        layout.addLayout(add_button_layout)
-
         self.__case_layout = QVBoxLayout()
-        self.__case_layout.setContentsMargins(0, 0, 0, 0)
-        layout.addLayout(self.__case_layout)
-
-        self.addCaseSelector(disabled=True)
-        layout.addStretch()
-
+        self.__case_layout.setSpacing(0)
+        self.__case_layout.setAlignment(Qt.AlignmentFlag.AlignTop)
+        scrollarea = QScrollArea()
+        scrollarea.setWidgetResizable(True)
+        button_layout_widget = QWidget()
+        button_layout_widget.setLayout(self.__case_layout)
+        scrollarea.setWidget(button_layout_widget)
+        self._addCheckButtons()
+        layout.addWidget(scrollarea)
         self.setLayout(layout)
 
-        self.__signal_mapper.mapped[QWidget].connect(self.removeWidget)
-
-    def __caseName(self, widget) -> str:
-        return str(self.__case_selectors[widget].currentText())
-
-    def getPlotCaseNames(self):
-        if self.__model.rowCount() == 0:
-            return []
-
-        return [self.__caseName(widget) for widget in self.__case_selectors_order]
-
-    def checkCaseCount(self):
-        self.__add_case_button.setEnabled(
-            len(self._cases) > 0 and len(self.__case_selectors_order) < 5
+    def getPlotCaseNames(self) -> List[str]:
+        return [widget.text() for widget in self.__button_list if widget.isChecked()]
+
+    def _addCheckButtons(self):
+        for case in self._cases:
+            button = CaseSelectCheckButton(
+                text=case, checkbutton_group=self.__button_list, parent=self
+            )
+            button.checkStateChanged.connect(self.caseSelectionChanged.emit)
+            self.__case_layout.insertWidget(0, button)
+            button.setMinimumWidth(20)
+            self.__button_list.append(button)
+
+        if len(self.__button_list) > 0:
+            self.__button_list[-1].setChecked(True)
+
+
+class CaseSelectCheckButton(QPushButton):
+    checkStateChanged = Signal()
+
+    def __init__(
+        self,
+        text="",
+        parent=None,
+        checkbutton_group: List["CaseSelectCheckButton"] = None,
+    ):
+        super(CaseSelectCheckButton, self).__init__(text=text, parent=parent)
+        if checkbutton_group is None:
+            checkbutton_group = []
+        self._checkbutton_group = checkbutton_group
+        self.setObjectName("case_selector")
+        self.setCheckable(True)
+
+    def nextCheckState(self):
+        if (self.isChecked() and not self._verifyCanUncheck()) or (
+            not self.isChecked() and not self._verifyCanCheck()
+        ):
+            return
```

Comment:
Would it be possible to invert the logic here?
```
if (not self.isChecked() and self. _verifyCanCheck()) or
    self.IsChecked() and self._verifyCanUnCheck())
```

Reply:
**Flashback to discreet mathematics**
Probably. I am currently doing cleanup.

Reply:
I think I improved it a little bit! Can you see if it can be improved further? Splitting them into two separate if statements might improve readability.

Reply:
What I meant was rewriting this to avoid the return, meaning if-something-do-something as opposed to if-something-dont.

Reply:
Is it not better to have an early return, opposed to having all the logic in an if indentation?

Reply:
I guess this is a matter of style preference. Personally I'd opt for the IFTT variant, -- there might be no correct answer.